content/300-guides/400-migrate-to-prisma/03-migrate-from-mongoose.mdx

```diff
@@ -0,0 +1,1056 @@
+}
+
+model posts {
+  id           String  @id @default(auto()) @map("_id") @db.ObjectId
+  title        String
+  content      String
+  published    Boolean
+  v            Int     @map("__v")
+
+  author       users  @relation(fields: [authorId], references: [id])
+  authorId     String @map("author") @db.ObjectId
+
+-  categories   String[] @db.ObjectId
++  categories   categories[] @relation(fields: [categoryIds], references: [id])
++  categoryIds String[]     @db.ObjectId
+}
+
+model users {
+  id      String        @id @default(auto()) @map("_id") @db.ObjectId
+  v       Int           @map("__v")
+  email   String        @unique(map: "email_1")
+  name    String
+  profile UsersProfile?
+  posts   posts[]
+}
+```
+
+</tab>
+
+<tab>
+
+```prisma
+type UsersProfile {
+  bio String
+}
+
+model categories {
+  id   String @id @default(auto()) @map("_id") @db.ObjectId
+  name String
+  v    Int    @map("__v")
+
+  posts    posts[]  @relation(fields: [postIds], references: [id])
+  postIds String[] @db.ObjectId
+}
+
+model posts {
+  id        String  @id @default(auto()) @map("_id") @db.ObjectId
+  title     String
+  content   String
+  published Boolean
+  v         Int     @map("__v")
+
+  author   users  @relation(fields: [authorId], references: [id])
+  authorId String @map("author") @db.ObjectId
+
+  categories   categories[] @relation(fields: [categoryIds], references: [id])
+  categoryIds String[]     @db.ObjectId
+}
+
+model users {
+  id      String        @id @default(auto()) @map("_id") @db.ObjectId
+  v       Int           @map("__v")
+  email   String        @unique(map: "email_1")
+  name    String
+  profile UsersProfile?
+  posts   posts[]
+}
+```
+
+</tab>
+
+</TabbedContent>
+
+### 2.5 Adjust the Prisma schema (optional)
+
+The models that were generated via introspection currently _exactly_ map to your database collections. In this section, you'll learn how you can adjust the naming of the Prisma models to adhere to [Prisma's naming conventions](/reference/api-reference/prisma-schema-reference#naming-conventions).
+
+Some of these adjustment are entirely optional and you are free to skip to the next step already if you don't want to adjust anything for now. You can go back and make the adjustments at any later point.
+
+As opposed to the current snake_case notation of Prisma models, Prisma's naming conventions are:
+
+- PascalCase for model names
+- camelCase for field names
+
+You can adjust the naming by _mapping_ the Prisma model and field names to the existing table and column names in the underlying database using `@@map` and `@map`.
+
+<Tip>
+
+You can use the [rename symbol](https://code.visualstudio.com/docs/editor/refactoring#_rename-symbol) operation to refactor model names by by highlighting the model name, pressing <kbd>F2</kbd>, and finally typing the desired name. This will rename all instances the model name is referenced, add the `@@map()` attribute to the existing model with it's former name.
```

Comment:
```suggestion
You can use the [rename symbol](https://code.visualstudio.com/docs/editor/refactoring#_rename-symbol) operation to refactor model names by highlighting the model name, pressing <kbd>F2</kbd>, and finally typing the desired name. This will rename all instances where it is referenced and add the `@@map(...)` attribute to the existing model with its former name.
```